mkosi/__init__.py

```diff
@@ -6791,24 +6786,62 @@ def cache_image_path(config: MkosiConfig, is_final_image: bool) -> Optional[Path
         return Path(f"{config.output}.{suffix}")
 
 
-def check_output(config: MkosiConfig) -> None:
+def check_tree_input(path: Optional[Path]) -> None:
+    # Each path may be a directory or a tarball.
+    # Open the file or directory to simulate an access check.
+    # If that fails, an exception will be thrown.
+    if not path:
+        return
+
+    os.open(path, os.R_OK)
+
+
+def check_script_input(path: Optional[Path]) -> None:
+    if not path:
+        return
+
+    os.open(path, os.R_OK)
+    if not path.is_file():
+        raise OSError(errno.ENOENT, 'Not a normal file')
+    if not os.access(path, os.X_OK):
+        raise OSError(errno.ENOENT, 'Not executable')
+    return None
+
+
+def check_inputs(config: MkosiConfig) -> None:
+    try:
+        for tree in (config.skeleton_trees,
+                     config.extra_trees):
+            for item in tree:
+                check_tree_input(item)
+
+        for path in (config.build_script,
+                     config.prepare_script,
+                     config.postinst_script,
+                     config.finalize_script,
+                     config.base_image):
+            check_script_input(path)
+    except OSError as e:
+        die(f'{e.filename} {e.strerror}')
+
+
+def check_outputs(config: MkosiConfig) -> None:
     if config.skip_final_phase:
         return
 
     for f in (
-        config.output,
-        config.output_checksum if config.checksum else None,
-        config.output_signature if config.sign else None,
-        config.output_bmap if config.bmap else None,
-        config.output_nspawn_settings if config.nspawn_settings is not None else None,
-        config.output_root_hash_file if config.verity else None,
-        config.output_sshkey if config.ssh else None,
-        config.output_split_root if config.split_artifacts else None,
-        config.output_split_verity if config.split_artifacts else None,
-        config.output_split_verity_sig if config.split_artifacts else None,
-        config.output_split_kernel if config.split_artifacts else None,
+            config.output,
+            config.output_checksum if config.checksum else None,
+            config.output_signature if config.sign else None,
+            config.output_bmap if config.bmap else None,
+            config.output_nspawn_settings if config.nspawn_settings is not None else None,
+            config.output_root_hash_file if config.verity else None,
+            config.output_sshkey if config.ssh else None,
+            config.output_split_root if config.split_artifacts else None,
+            config.output_split_verity if config.split_artifacts else None,
+            config.output_split_verity_sig if config.split_artifacts else None,
+            config.output_split_kernel if config.split_artifacts else None,
```

Comment:
Is there a reason to add indentation here? It's a lot of churn for at most a small gain in readability.

Reply:
I dropped this chunk now.